resources/gnomad_liftover.py

```diff
@@ -0,0 +1,160 @@
+    :param ReferenceGenome rg: Reference genome, b38
+    :return: gnomAD release Table lifted to b38
+    :rtype: Table
+    """
+
+    # get reference genomes and load liftover chain file
+    #rg37 = hl.get_reference('GRCh37')
+    #rg38 = rg38 = hl.get_reference('GRCh38')
+    #rg37.add_liftover('gs://hail-common/references/grch37_to_grch38.over.chain.gz', rg38)
+
+    # annotate release file with liftover coordinates
+    ht = ht.annotate(new_locus = hl.liftover(ht.locus, rg, include_strand=True),
+                     old_locus=ht.locus
+                    )
+    ht = ht.key_by(locus = ht.new_locus.result)
+    ht = ht.annotate(no_target = hl.is_defined(ht.locus))
+    ht = checkpoint(ht, dtype)
+    return ht
+
+
+def reverse_complement(allele: str):
+    """
+    Returns the reverse complement of an allele
+    :param str allele: Allele to be flipped
+    :return: Reverse complement of input allele
+    :rtype: str
+    """
+ 
+    return (hl.switch(allele)
+            .when('A', 'T')
+            .when('T', 'A')
+            .when('G', 'C')
+            .when('C', 'G')
+            .default(allele))
+
+
+def snp_mismatch(ht: hl.Table, dtype: str, rg: hl.genetics.ReferenceGenome) -> hl.Table:
+    """
+    Annotates mismatches between reference allele and allele in reference fasta
+    :param Table ht: Table of SNPs to be annotated
+    :param str dtype: Data type (exome or genome)
+    :param ReferenceGenome rg: GRCh38 reference genome with fasta sequence loaded
+    :return: Table annotated with mismatches between reference allele and allele in fasta
+    :rtype: Table
+    """
+ 
+    # filter to snps
+    ht = ht.filter(hl.is_snp(ht.alleles[0], ht.alleles[1]))
+    snp_count = ht.count()
+    logger.info('{} SNPs in table'.format(snp_count))
+
+    # check if reference allele matches what is in reference fasta
+    # for snps on negative strand, make sure reverse complement of ref allele matches what is in reference fasta
+    ht = ht.annotate(
+        reference_mismatch = hl.cond(
+                                ht.new_locus.is_negative_strand,
+                                (reverse_complement(ht.alleles[0]) != hl.get_sequence(ht.locus.contig, ht.locus.position, reference_genome=rg)),
+                                (ht.alleles[0] != hl.get_sequence(ht.locus.contig, ht.locus.position, reference_genome=rg))
+                                )
+                )
+
+    #ht = checkpoint(ht, dtype)
+    return ht
+ 
+def check_mismatch(ht: hl.Table) -> dict:
+    """
+    Checks for mismatches between reference allele and allele in reference fasta
+    :param Table ht: Table to be checked
+    :return: Dictionary containing mismatch type ('total' or 'minus' as key) and umber of mismatches as value
+    :rtype: dict
+    """
+
+    mismatch = {}
+    ht = ht.filter(ht.reference_mismatch)
+    mismatch['total'] = ht.count()
+    mismatch['minus'] =  ht.filter(ht.new_locus.is_negative_strand).count()
```

Comment:
If you're finding this takes a while, it might be easier to put these queries (and more - you could also simultaneously count the ones that are not mismatched by removing the filter above) into one big aggregate. Let me know if that makes sense or I can suggest further.